src/BSON/Decimal128.c

```diff
@@ -148,24 +218,38 @@ phongo_create_object_retval php_phongo_decimal128_create_object(zend_class_entry
 #endif
 } /* }}} */
 
-HashTable *php_phongo_decimal128_get_debug_info(zval *object, int *is_temp TSRMLS_DC) /* {{{ */
+HashTable *php_phongo_decimal128_get_properties(zval *object TSRMLS_DC) /* {{{ */
 {
 	php_phongo_decimal128_t *intern;
-	char                  outbuf[BSON_DECIMAL128_STRING] = "";
-#if PHP_VERSION_ID >= 70000
-	zval                  retval;
-#else
-	zval                  retval = zval_used_for_init;
-#endif
+	HashTable               *props;
+	char                     outbuf[BSON_DECIMAL128_STRING] = "";
 
 	intern = Z_DECIMAL128_OBJ_P(object);
-	*is_temp = 1;
-	array_init_size(&retval, 1);
+	props = zend_std_get_properties(object TSRMLS_CC);
+
+	/* TODO: track if intern is uninitialized and return early. This may require
+	 * creating an extra field, since intern->decimal is not a pointer */
```

Comment:
@derickr: Let's decide how best to handle this. I left similar todo items in the Timestamp and UTCDateTime classes as well.